math/utils.go

```diff
@@ -53,6 +53,56 @@ func CalcEma(
 	return ret, nil
 }
 
+// NCalcEma is a generalisation of calculating the EMA update n times.
+func NCalcEma(
+	alpha,
+	update,
+	score Dec,
```

Comment:
terms should be more general than this e.g. currentVal, updatedVal. This may not always apply to scores, as this is a general Dec